--- origsrc/waf-1.5.8/test/ccroot_test.py	2009-06-11 12:52:22.000000000 -0500
+++ src/waf-1.5.8/test/ccroot_test.py	2009-07-19 18:02:34.448792500 -0500
@@ -203,9 +203,12 @@ def configure(conf):
 		else:
 			self.assert_(env['shlib_CXXFLAGS'] == [], 'incorrect shlib CXXFLAGS was %s' % env['shlib_CXXFLAGS'])
 
-		if target_platform == 'win32' or target_platform == 'cygwin':
+		if target_platform == 'win32':
 			self.assert_(env['program_PATTERN'] == '%s.exe', 'incorrect program pattern, was %s, target_platform = %s' % (env['program_PATTERN'], target_platform))
 			self.assert_(env['shlib_PATTERN'] == '%s.dll', 'incorrect shlib pattern')
+		elif target_platform == 'cygwin':
+			self.assert_(env['program_PATTERN'] == '%s.exe', 'incorrect program pattern, was %s, target_platform = %s' % (env['program_PATTERN'], target_platform))
+			self.assert_(env['shlib_PATTERN'] == 'cyg%s.dll', 'incorrect shlib pattern')
 		elif target_platform == 'linux2':
 			self.assert_(env['program_PATTERN'] == '%s', 'incorrect program pattern')
 			self.assert_(env['shlib_PATTERN'] == 'lib%s.so', 'incorrect shlib pattern')
