http://code.google.com/p/waf/issues/detail?id=495

--- origsrc/waf/demos/cxx/src/wscript	2009-07-20 11:12:58.000000000 -0500
+++ src/waf/demos/cxx/src/wscript	2009-07-20 11:38:42.191147800 -0500
@@ -36,7 +36,7 @@ def build(bld):
 
 	# 4. Another shared library
 	bld.new_task_gen(
-		features = 'cxx cshlib',
+		features = 'cxx cshlib implib',
 		source = 'd1.cpp',
 		target = 'shlib1',
 		name = 'john',
--- origsrc/waf/demos/swig/wscript	2009-07-20 11:13:04.000000000 -0500
+++ src/waf/demos/swig/wscript	2009-07-20 11:42:24.111039700 -0500
@@ -47,7 +47,7 @@ def build(bld):
 	# to the source of extend/python/_test_swig_waf and remeve uselib_local
 
 	bld.new_task_gen(
-		features = 'cxx cshlib',
+		features = 'cxx cshlib implib',
 		source = 'extend/a.cpp',
 		target = 'extend/mylib',
 		includes = 'extend',
