--- origsrc/waf-1.5.18/wafadmin/Tools/vala.py	2010-07-06 12:50:48.000000000 -0500
+++ src/waf-1.5.18/wafadmin/Tools/vala.py	2010-09-13 03:32:49.106636000 -0500
@@ -118,7 +118,7 @@ class valac_task(Task.Task):
 						api_version = version[0] + ".0"
 				install_path = '${INCLUDEDIR}/%s-%s/%s' % (package, api_version, header.relpath_gen(top_src))
 				bld.install_as(install_path, header, self.env)
-			bld.install_files('${DATAROOTDIR}/vala/vapi', vapi_list, self.env)
+			bld.install_files('${DATAROOTDIR}/vala-0.10/vapi', vapi_list, self.env)
 			bld.install_files('${DATAROOTDIR}/gir-1.0', gir_list, self.env)
 
 	def _fix_output(self, output):
