--- origsrc/waf-1.5.16/wafadmin/3rdparty/boost.py	2010-05-12 21:25:00.221536700 -0500
+++ src/waf-1.5.16/wafadmin/3rdparty/boost.py	2010-05-12 21:25:01.625617000 -0500
@@ -253,7 +253,7 @@ def find_boost_library(self, lib, kw):
 	(libname, file) = (None, None)
 	if kw['static'] in [STATIC_NOSTATIC, STATIC_BOTH]:
 		st_env_prefix = 'LIB'
-		files = libfiles(lib, v['shlib_PATTERN'], lib_paths)
+		files = libfiles(lib, v['implib_PATTERN'], lib_paths)
 		(libname, file) = find_library_from_list(lib, files)
 	if libname is None and kw['static'] in [STATIC_ONLYSTATIC, STATIC_BOTH]:
 		st_env_prefix = 'STATICLIB'
